src/runtime/base.rs

```diff
@@ -304,6 +304,9 @@ pub struct StatusError {
     status: Status,
 }
 
+unsafe impl Send for StatusError {}
+unsafe impl Sync for StatusError {}
```

Comment:
I am not sure if this is safe. How are you sure if `iree_status_t` is thread-safe?